core/app/finders/spree/option_values/find_filterable.rb

```diff
@@ -0,0 +1,24 @@
+module Spree
+  module OptionValues
+    class FindFilterable
+      def initialize(scope: OptionValue.spree_base_scopes, products_scope: Product.spree_base_scopes)
+        @scope = scope
+        @products_scope = products_scope
+      end
+
+      def execute
+        scope.
+          filterable.
+          where(variants: { product_id: products_ids })
+      end
+
+      private
+
+      attr_reader :scope, :products_scope
+
+      def products_ids
+        products_scope.map(&:id)
```

Comment:
I used `map(&:id)` instead of `ids` due to an issue with `distinct`. Basically, Postgres requires any column that's used to order to be in `SELECT DISTINCT`, too.

When using a combination of `in_taxon` and `with_currency` scopes we do 2 things:
- order by taxon position
- select distinct products

Then, when we only want ids from products Postgres raises an error, requiring also taxon position in the select clause.